new-lamassu-admin/src/pages/Triggers/Triggers.js

```diff
@@ -30,82 +26,71 @@ const SAVE_CONFIG = gql`
   }
 `
 
-const GET_INFO = gql`
+const GET_CONFIG = gql`
   query getData {
     config
   }
 `
 
+const GET_CUSTOM_REQUESTS = gql`
+  query customInfoRequests {
+    customInfoRequests {
+      id
+      customRequest
+      enabled
+    }
+  }
+`
+
 const Triggers = () => {
   const classes = useStyles()
   const [wizardType, setWizard] = useState(false)
   const [showCustomInfoRequests, setShowCustomInfoRequests] = useState(false)
-  const { data, loading } = useQuery(GET_INFO)
-  const triggers = fromServer(data?.config?.triggers ?? [])
+  const { data, loading } = useQuery(GET_CONFIG)
+  const { data: customInfoReqData } = useQuery(GET_CUSTOM_REQUESTS)
+
+  const customInfoRequests =
+    R.path(['customInfoRequests'])(customInfoReqData) ?? []
+  const enabledCustomInfoRequests = R.filter(R.propEq('enabled', true))(
+    customInfoRequests
+  )
 
+  const triggers = fromServer(data?.config?.triggers ?? [])
   const complianceConfig =
     data?.config && fromNamespace('compliance')(data.config)
   const rejectAddressReuse = complianceConfig?.rejectAddressReuse ?? false
-  const [error, setError] = useState(null)
 
   const [saveConfig] = useMutation(SAVE_CONFIG, {
     onCompleted: () => setWizard(false),
-    refetchQueries: () => ['getData'],
-    onError: error => setError(error)
+    refetchQueries: () => ['getData']
   })
 
-  const add = rawConfig => {
-    const toSave = R.concat([{ id: v4(), direction: 'both', ...rawConfig }])(
-      triggers
-    )
-    return saveConfig({ variables: { config: { triggers: toServer(toSave) } } })
-  }
-
   const addressReuseSave = rawConfig => {
     const config = toNamespace('compliance')(rawConfig)
     return saveConfig({ variables: { config } })
   }
 
-  const save = config => {
-    setError(null)
-    return saveConfig({
-      variables: { config: { triggers: toServer(config.triggers) } }
-    })
-  }
-
-  const currency = R.path(['fiatCurrency'])(
-    fromNamespace(namespaces.LOCALE)(data?.config)
-  )
-
   const titleSectionWidth = {
     [classes.tableWidth]: !showCustomInfoRequests
   }
 
-  const [customRequests, setCustomRequests] = useState([])
-
-  const alter = R.curry((values, key, items) => {
-    const merge = R.mergeLeft(values)
-    return R.map(R.when(R.propEq('id', key), merge), items)
-  })
-
-  const handleDeleteCustomRequest = req => {
-    return setCustomRequests(R.reject(o => o.id === req.id)(customRequests))
+  const setBlur = shouldBlur => {
+    return shouldBlur
+      ? document.querySelector('#root').classList.add('root-blur')
+      : document.querySelector('#root').classList.remove('root-blur')
   }
 
-  const handleSaveCustomRequest = (values, isEditing) => {
-    if (!isEditing) {
-      return setCustomRequests([
-        ...customRequests,
-        { id: customRequests.length, ...values }
-      ])
+  const toggleWizard = wizardName => forceDisable => {
```

Comment:
Not the biggest fan of having one toggle function handling multiple wizards, but shouldn't be a problem if properly handled. What I generally use is a toggle for each popup, but I believe that loads all components (even if they are off) onto the DOM, which is not a good idea.

One thing I'd like to try on with React related to popups, modals, tooltips, etc. are portals. In case you're interested https://reactjs.org/docs/portals.html

Reply:
I didn't see it as one function for multiple wizards, other than reusing the same state variable
I looked at this like a function that returns a function, the first function "builds" the second function for the wizard we want